device/samples/pnp/pnpTemperatureController.js

```diff
@@ -0,0 +1,256 @@
+//  'HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>'
+const deviceConnectionString = process.env.DEVICE_CONNECTION_STRING;
+
+const modelId = 'dtmi:com:example:TemperatureController;1';
+const messageSubjectProperty = '$.sub';
+const thermostat1ComponentName = 'thermostat1';
+const thermostat2ComponentName = 'thermostat2';
+const deviceInfoComponentName = 'deviceInformation';
+const commandComponentCommandNameSeparator = '*';
+let intervalToken1;
+let intervalToken2;
+let intervalToken3;
+
+const commandNameGetMaxMinReport1 = thermostat1ComponentName + commandComponentCommandNameSeparator + 'getMaxMinReport';
+const commandNameGetMaxMinReport2 = thermostat2ComponentName + commandComponentCommandNameSeparator + 'getMaxMinReport';
+const commandNameReboot = 'reboot';
+
+const commandHandler = async (request, response) => {
+  helperLogCommandRequest(request);
+  switch (request.methodName) {
+  case commandNameGetMaxMinReport1: {
+    await sendCommandResponse(request, response, 200, 'max min report from thermostat 1');
+    break;
+  }
+  case commandNameGetMaxMinReport2: {
+    await sendCommandResponse(request, response, 200, 'max min report from thermostat 2');
+    break;
+  }
+  case commandNameReboot: {
+    await sendCommandResponse(request, response, 200, 'reboot response');
+    break;
+  }
+  default:
+    await sendCommandResponse(request, response, 404, 'unknown method');
+    break;
+  }
+};
+
+const sendCommandResponse = async (request, response, status, payload) => {
+  try {
+    await response.send(status, payload);
+    console.log('Response to method \'' + request.methodName +
+              '\' sent successfully.' );
+  } catch (err) {
+    console.error('An error ocurred when sending a method response:\n' +
+              err.toString());
+  }
+};
+
+const helperLogCommandRequest = (request) => {
+  console.log('Received command request for comand name \'' + request.methodName + '\'');
+
+  if (!!(request.payload)) {
+    console.log('The command request paylaod :');
+    console.log(request.payload);
+  }
+};
+
+
+const helperCreateReportedPropertiesPatch = (propertiesToReport, componentName) => {
+    let patch;
+    if (!!(componentName)) {
+      patch = { };
+      propertiesToReport.__t = "c";
+      patch[componentName] = propertiesToReport;
+    }
+    else {
+      patch = { };
+      patch = propertiesToReport;
+      }
+    if (!!(componentName)) {
+    console.log('The following properties will be updated for component:' + componentName);
+    }
+    else{
+      console.log('The following properties will be updated for root interface:');
+    }
+    console.log(patch);
+    return patch;
+  };
+
+const updateComponentReportedProperties = (deviceTwin, patch, componentName) => {
+  let logLine;
+  if (!!(componentName))
+  {
+    logLine = 'Properties have been reported for component:' + componentName;
+  }
+  else{
```

Comment:
>else{ [](start = 2, length = 5)

Space needed after the else.  Generally in the node SDK we put the else on the same line as the closing brace for the if.  But you are consistent here so ....

Reply:
same reason of linting